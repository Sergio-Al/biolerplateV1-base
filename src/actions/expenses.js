import { v4 as uuidv4 } from "uuid";
import db from "../firebase/firebase";
import { ref, push } from "firebase/database";

// ADD_EXPENSE
export const addExpense = (expense) => ({
  type: "ADD_EXPENSE",
  expense,
});

export const startAddExpense = (expenseData = {}) => {
  // we are returning a function that redux will exetutes
  // if is a traditional return function (not for redux) this could not work as expected
  // this is called internally by redux with dispatch as parameter (redux puts a dispatch as parameter in our return funcion)
  // this return function executes firebase operation first then dispatch to our redux store
  return (dispatch) => {
    const {
      description = "",
      note = "",
      amount = 0,
      createdAt = 0,
    } = expenseData;
    const expense = { description, note, amount, createdAt };

    return push(ref(db, "expenses"), expense).then((ref) => {
      dispatch(
        addExpense({
          id: ref.key,
          ...expense,
        })
      );
    });
  };
};

// REMOVE_EXPENSE
export const removeExpense = ({ id } = {}) => ({
  type: "REMOVE_EXPENSE",
  id,
});
// EDIT_EXPENSE
export const editExpense = (id, updates) => ({
  type: "EDIT_EXPENSE",
  id,
  updates,
});
