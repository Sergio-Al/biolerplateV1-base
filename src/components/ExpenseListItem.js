import React from "react";
import { Link } from "react-router-dom";

const ExpenseListItem = ({ id, description, amount, createdAt }) => (
  <div>
    <h3>
      Description: {description}
      <Link to={`/edit/${id}`}> Edit</Link>
    </h3>
    <p>
      {amount} - {createdAt}
    </p>
  </div>
);

export default ExpenseListItem;
