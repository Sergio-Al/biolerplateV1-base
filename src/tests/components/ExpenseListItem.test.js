import React from "react";
import { shallow } from "enzyme";
import ExpenseListItem from "../../components/ExpenseLIstItem";
import expenses from "../fixtures/expenses";

test("should render expense List item", () => {
  const wrapper = shallow(<ExpenseListItem {...expenses[1]} />);
  expect(wrapper).toMatchSnapshot();
});
