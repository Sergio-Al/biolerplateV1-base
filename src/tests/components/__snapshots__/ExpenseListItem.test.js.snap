// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render expense List item 1`] = `
<div>
  <h3>
    Description: 
    Rent
    <[object Object]
      to="/edit/2"
    >
       Edit
    </[object Object]>
  </h3>
  <p>
    $1,095.00
    -
    December 27th, 1969
  </p>
</div>
`;
